apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.Tests/StorageClientImplTest.cs

```diff
@@ -69,5 +76,402 @@ public void GetEffectiveEncryptionKey_Invalid(EncryptionKey clientkey, Encryptio
             var exception = Assert.Throws<ArgumentException>(() => client.GetEffectiveEncryptionKey(operationKey, kmsKey));
             Assert.Contains(expectedMessageSubstring, exception.Message);
         }
+
+        // Tests for GCS Retry :
+        // - No retry applied
+        // - Retry then success (sometimes with conditions)
+        // - Retry with eventual failure
+
+
+        private static void NoRetryHelper<T>(
```

Comment:
I would suggest moving all the helper methods to the end of the class, and adding an XML comment for each one.